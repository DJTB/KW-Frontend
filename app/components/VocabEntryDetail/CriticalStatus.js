import React from 'react';
import PropTypes from 'prop-types';

import H5 from 'base/H5';
import Element from 'base/Element';

CriticalStatus.propTypes = {
  isCritical: PropTypes.bool.isRequired,
};

function CriticalStatus({ isCritical }) {
  return isCritical && <Element><H5>Critical!</H5></Element>;
}

export default CriticalStatus;
