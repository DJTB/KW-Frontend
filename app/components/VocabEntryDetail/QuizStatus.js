import React from 'react';
import PropTypes from 'prop-types';

import P from 'base/P';
import Element from 'base/Element';

QuizStatus.propTypes = {
  text: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
};

function QuizStatus({ text, status }) {
  return (
    <Element>
      <P><b>{text}</b>: {status}</P>
    </Element>
  );
}

export default QuizStatus;
