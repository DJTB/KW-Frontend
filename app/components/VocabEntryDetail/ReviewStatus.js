import React from 'react';
import PropTypes from 'prop-types';
import getDateInWords from 'utils/getDateInWords';

import P from 'base/P';
import Element from 'base/Element';

ReviewStatus.propTypes = {
  isHidden: PropTypes.bool,
  isReviewReady: PropTypes.bool,
  nextReviewDate: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.instanceOf(Date),
  ]),
};

ReviewStatus.defaultProps = {
  isHidden: false,
  isReviewReady: false,
  nextReviewDate: false,
};

function ReviewStatus({ isHidden, isReviewReady, nextReviewDate }) {
  let status = 'N/A';
  if (isHidden) { status = 'Hidden'; }
  if (isReviewReady) { status = 'Now'; }
  if (nextReviewDate) { status = getDateInWords(nextReviewDate); }
  return (
    <Element style={{ flex: '1 1 100%' }}>
      <P><b>Next review: </b>{status}</P>
    </Element>
  );
}

export default ReviewStatus;
