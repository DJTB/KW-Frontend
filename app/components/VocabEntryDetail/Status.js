import React from 'react';
import PropTypes from 'prop-types';

import P from 'base/P';
import Element from 'base/Element';

Status.propTypes = {
  text: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
};

function Status({ text, status }) {
  return (
    <Element style={{ flex: '1 1 100%' }}>
      <P><b>{text}</b>: {status}</P>
    </Element>
  );
}

export default Status;
