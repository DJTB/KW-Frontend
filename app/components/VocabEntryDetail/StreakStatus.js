import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import getSrsRankName from 'utils/getSrsRankName';
import StreakIcon from 'components/StreakIcon';

import H5 from 'base/H5';

const Wrapper = styled.span`
  &:first-of-type {
    margin-right: .75rem;
  }
`;
const Text = H5.extend`
  display: inline-flex;
`;

StreakStatus.propTypes = {
  streak: PropTypes.number.isRequired,
  category: PropTypes.string.isRequired,
};

function StreakStatus({ streak, category }) {
  return (
    <Wrapper>
      <Text>{category}: </Text>
      <StreakIcon streakName={getSrsRankName(streak)} size="1.75em" />
    </Wrapper>
  );
}


export default StreakStatus;
