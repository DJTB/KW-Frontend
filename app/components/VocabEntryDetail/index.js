import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';

import PrimaryReading from 'components/PrimaryReading';
import ReadingHeader from 'components/ReadingHeader';
import SynonymHeader from 'components/SynonymHeader';
import SentencePair from 'components/SentencePair';

VocabEntryDetail.propTypes = {
  entry: PropTypes.object.isRequired,
};

/* eslint-disable react/prop-types */
const renderReading = ({ character, kana, sentenceEn, sentenceJa }) => (
  <div>
    <PrimaryReading
      character={character}
      kana={kana}
    />
    <SentencePair
      sentenceEN={sentenceEn}
      sentenceJA={sentenceJa}
      character={character}
      kana={kana}
    />
  </div>
);
/* eslint-enable */

function VocabEntryDetail({ entry }) {
  return (
    <div>
      {entry.readings.slice(0, 1).map((reading) => renderReading(reading))}
      {entry.readings.slice(1).map((reading) => (
        <div key={uuid}>
          <ReadingHeader
            id={entry.id}
            primaryCharacter={reading.character}
            tags={reading.tags}
          />
          {renderReading(reading)}
        </div>
      ))}
      {entry.synonyms.map((synonym) => (
        <div key={uuid}>
          <SynonymHeader tags={synonym.tags} />
          {renderReading(synonym)}
        </div>
      ))}
      <p>Your Progression: WK: Burned KW: Guru</p>
      <p>Answered Correct: 95% of 23 times</p>


      <p>debug</p>
      <pre>
        <code>
          {JSON.stringify(entry, null, 2)}
        </code>
      </pre>
    </div>
  );
}

export default VocabEntryDetail;
