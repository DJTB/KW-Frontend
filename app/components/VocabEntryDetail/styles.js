import styled from 'styled-components';

import { gutter } from 'shared/styles/layout';

import { Form as VocabEntryNotes } from 'components/VocabEntryNotes/styles';
import StreakStatus from './StreakStatus';
import Status from './Status';
import CriticalStatus from './CriticalStatus';
import ReviewStatus from './ReviewStatus';
import QuizStatus from './QuizStatus';

export const Wrapper = styled.div`
  ${gutter()};
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  & > ${VocabEntryNotes} {
    flex: 1 0 100%;
  }
`;

export const Text = styled.span`
  ${gutter()};
`;
