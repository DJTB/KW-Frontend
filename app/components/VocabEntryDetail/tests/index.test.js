import React from 'react';
import { shallow } from 'enzyme';
import VocabEntryDetail from '../index';

describe('<VocabEntryDetail />', () => {
  it('should match baseline snapshot', () => {
    expect(shallow(<VocabEntryDetail />)).toMatchSnapshot();
  });
});
