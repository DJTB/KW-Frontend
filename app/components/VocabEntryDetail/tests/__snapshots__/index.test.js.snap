// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VocabEntryDetail /> should match baseline snapshot 1`] = `
<div>
  <p>
    Entry
  </p>
  <pre>
    <code />
  </pre>
  <p>
    Primary
  </p>
  <pre>
    <code />
  </pre>
</div>
`;
